bdk_cbf_example/src/cbf.rs

```diff
@@ -0,0 +1,166 @@
+        let domains = match domains {
+            Domains::IPv4 => vec![Domain::IPV4],
+            Domains::IPv6 => vec![Domain::IPV6],
+            Domains::Both => vec![Domain::IPV4, Domain::IPV6],
+        };
+
+        let config = Config {
+            network,
+            domains,
+            ..Config::default()
+        };
+
+        let client = Client::<Reactor>::new()?;
+        let handle = client.handle();
+
+        // Run the client on a different thread, to not block the main thread.
+        // Note that when we sync we rescan from the latest point of agreement,
+        // so it's not a problem if the client is running in the background
+        thread::spawn(|| client.run(config).unwrap());
+
+        Ok(CbfClient {
+            handle,
+            setup: None,
+        })
+    }
+
+    pub fn sync_setup(
+        &mut self,
+        scripts: impl Iterator<Item = Script>,
+        processed_height: Option<BlockId>,
+    ) -> anyhow::Result<()> {
+        println!("Looking for peers...");
+        self.handle.wait_for_peers(1, Services::default())?;
+        println!("Connected to at least one peer");
+
+        let events = self.handle.events();
+        let blocks_matched = HashSet::new();
+        let peer_height = None;
+
+        self.setup = Some(CbfSetup {
+            events,
+            processed_height,
+            blocks_matched,
+            peer_height,
+        });
+
+        let processed_height = processed_height.map(|h| h.height as u64).unwrap_or(0);
+
+        // TODO: maybe we should check our latest point of agreement?
+        self.handle.rescan(processed_height.., scripts)?;
+        println!("Rescanning chain from height {:?}", processed_height);
+        Ok(())
+    }
+
+    pub fn next_event(&mut self) -> anyhow::Result<CbfEvent> {
+        if self.setup.is_none() {
+            return Err(anyhow!("Need to call sync_setup first".to_string()));
+        }
+        let mut setup = self.setup.as_mut().expect("We check above");
+        loop {
+            chan::select! {
+                recv(setup.events) -> event => {
+                    let event = event?;
+                    match event {
+                        Event::PeerNegotiated { height, .. } => {
+                            println!("* peer negotiated with height {:?}", height);
+                            let height = height as u32;
+                            if setup.peer_height < Some(height) {
+                                setup.peer_height = Some(height);
+                            }
+                            if setup.processed_height.map(|h| h.height) == setup.peer_height && setup.blocks_matched.is_empty() {
+                                // TODO: improve this!
+                                // It might be that both me and this peer
+                                // are lagging behind
+                                return Ok(CbfEvent::ChainSynced(setup.processed_height));
```

Comment:
What about using nakamoto's `Event::Synced` to emit this? This seems very error prone and even an attack vector. I think you shouldn't need to keep track of `setup.peer_height` at all.

So when you get `Event::Synced` with height == `Handle::get_tip()` you can emit this.